### PR TITLE
Resolve several nits in the Terminal AI code

### PR DESCRIPTION
**Pull request by PankajBhojwani**
- [x] Remove the ESC handler that clears the query message
- [x] Streamline error handling code
- [x] Fix `this` and `&` in several lambdas
- [x] Fix getting the [active commandline properly](https://github.com/microsoft/terminal/pull/16285#discussion_r1396422350)
- [x] Use fwlinks
- [x] Use XAML color ramp resource names instead of hardcoded colors

### CODE REVIEW

**Comment by DHowett** [comment]
Would this be the right place to switch all of the hardcoded colors to use the standard XAML color ramp resource names? That would let us support light mode...

**Comment by PankajBhojwani** [comment]
> Would this be the right place to switch all of the hardcoded colors to use the standard XAML color ramp resource names? That would let us support light mode...

Yep! Added to the todo list

**Review by DHowett — CHANGES_REQUESTED** [review]
The weak reference juggling sorta complicates the code to protect us from actually-impossible scenarios. If we have seen these crashes in the wild, it's probably worth fixing them. If not, it makes the diff a lot bigger for little reason :)

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -5319,51 +5319,58 @@ namespace winrt::TerminalApp::implementation
...
-                        _extensionPalette.IconPath(profile.Icon());
-                        break;
-                    }
+                    page->_extensionPalette.ActiveCommandline(L"");
                 }
             }
-            else
+        });
+        _extensionPalette.AIKeyAndEndpointRequested([weakThis{ get_weak() }](IInspectable const&, IInspectable const&) {
+            if (auto page{ weakThis.get() })
```
[the remark this answers is not in the document]
Comment (PankajBhojwani) [reply src/cascadia/TerminalApp/TerminalPage.cpp:5369]: Happy to undo these changes! But I figured this was similar enough to `_newTabButton.Click` (which also uses a weak ref to the page) that it made sense for consistency.

**Comment by PankajBhojwani** [comment]
Dark mode:
image

Light mode:
image

High contrast (Desert):
image

High contrast (Night sky):
image

**Review by DHowett — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Pull request merged**